### PR TITLE
[PULL REQUEST] Improve detection of element with watchForElement

### LINKED ISSUE #1269: awaitElement helper is buggy
I have made this bypass on my local fork.

```js
this.helpers.awaitElement('#link1s[href]', a => this.helpers.safelyNavigate(a.href))
```

When I tested it, it's sometimes work, sometimes won't. Any tips for @NotAProton?

_Originally posted by @kiraio-moe in https://github.com/FastForwardTeam/FastForward/issues/40#issuecomment-1837472903_

### PR DESCRIPTION
**Pull request by kiraio-moe**
- [X] I made sure there are no unnecessary changes in the code;
- [X] Tested on Chromium (Includes Opera, Brave, Vivaldi, Edge, etc);
- [X] Tested on Firefox.

FIX #1269

### CODE REVIEW

File: src/helpers/dom.js [hunk 1]
```diff
@@ -175,6 +181,38 @@ export function awaitElement(elem, func) {
...
+      callback(element)
+      resolve(element)
+      return
+    }
+
+    const observer = new MutationObserver(() => {
+      Array.from(document.querySelectorAll(selector)).forEach(el => {
+        console.log(`[FastForward] Element found: ${el} | Selector: ${selector}`)
+        callback(el)
+        resolve(el)
```
Comment (NotAProton) [inline src/helpers/dom.js]: Make it promise-only
```suggestion
 * @returns {Promise<any>} A promise that resolves with the element.
 */
export function watchForElement (selector) {
  return new Promise(resolve => {
    const element = document.querySelector(selector)
    if (element) {
      console.log(`[FastForward] Element found: ${element} | Selector: ${selector}`)
      resolve(element)
      return
    }

    const observer = new MutationObserver(() => {
      Array.from(document.querySelectorAll(selector)).forEach(el => {
        console.log(`[FastForward] Element found: ${el} | Selector: ${selector}`)
        resolve(el)
```

**Review by NotAProton — CHANGES_REQUESTED** [review]
Sorry for the delay in reviewing

**Comment by kiraio-moe** [comment]
Sure, I'll update it on Sunday.

Pada Sab, 30 Des 2023 21.53, NotAProton ***@***.***> menulis:

File: src/helpers/dom.js (hunk 1 above)
Comment (kiraio-moe) [reply src/helpers/dom.js]: Ok, done.

**Review by NotAProton — APPROVED** [review]

**Pull request merged**